chore: trigger docs deploy

albttx opened the pull request:
This trigger the github action `netlify.yml` on [gnolang/docs.gno.land](https://github.com/gnolang/docs.gno.land) when a merge is done on master and the docs/ folder have updates.

I will need someone (cc: @moul ) to create and add a PAT to this repo secret 🙏🏼 

EDIT:

This PR: 

- Bring a dev env for the docs.
- Add github action to call gnolang/docs.gno.land netlify deploy action

I'm currently working on a netlify preview, but it's a bit more complexe than expected, it will come in another PR.

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

albttx commented on .github/workflows/deploy-docs.yml:
+ `rm -rf misc/docusaurus` ;)

leohhhn replied:
I don't think we should be removing `misc/docusaurus`, at least until we have a preview deployment of ongoing PRs. This is because this is currently the only way to preview how the change in the docs will look like while making it. Also, the sidebar ordering is done in the docusaurus folder.

albttx replied:
@leohhhn production has already been migrated to the repository gnolang/docs.gno.land.

So any changes in misc/docusaurus won't have any preview anymore. The search engine isn't developed here either.
IF there is missing stuff in gnolang/docs.gno.land, we must migrate them right now.

Keeping misc/docusaurus will only lead to confusions

leohhhn replied:
Here are two critical points regarding this PR:

1. Markdown content & docusaurus config are coupled. They both combine to make the docs the way they are.  
3. People need to be able to preview the changes they are making to the docs in the docusaurus environment, not only in a simple markdown render.

I suggest we keep the working versions of both `docs/` and `misc/docusaurus` in the monorepo, and on each merge to master a CI copies over both folders to the docs.gno.land repo, and deploys over there.

If there is another way to fulfill both of the points above, please let me know.

albttx replied:
Ok, I know how to handle that, I'll work on it!

zivkovicmilos replied:
@albttx 
What's the status on this?

albttx replied:
Still a work in progress, i made improvement today

leohhhn reviewed (changes requested):
@albttx Please reference the comment I left above before merging
https://github.com/gnolang/gno/pull/1801/files#r1532504955

zivkovicmilos commented:
@albttx What is the status of this PR?

albttx commented:
Hey @zivkovicmilos, if it's good for you, let's first merge this PR, as it is.
It will required to make a second PR on gnolang/docs.gno.land to update the sidebar.js file. 

I'll work on a solution on a separate pull request.

albttx commented:
link with : https://github.com/gnolang/docs.gno.land/pull/21

leohhhn commented:
Ok, so from what I understand:
- This PR adds a docker image to the `docs/` folder, that will pull the misc files from the docs.gno.land repo for a preview. This image can be used for a local preview and will be also used to run the CI preview on this repo.
- [This PR](https://github.com/gnolang/docs.gno.land/pull/21) adds the actual `dev` image to be pulled by the monorepo, and adds a prod image for the actual deployment of the docs.

@albttx can you please confirm that this is the setup?

moul commented on docs/sidebars.js:
no JS.

yaml, whatever, but no JS.

moul replied:
Just a reminder: The purpose of the docs folder is to be embedded in gnoweb. See https://github.com/gnolang/gno/issues/522#issuecomment-2010133490.

I suggest using simple filenames with prefixes, such as `000_hello_world.md`, `050_foo.md`, etc.

albttx replied:
I put a yaml file. 

Maybe we can in another PR think to have a different solution using the markdown header to gathers the informations + prefixes as you mentionned.

moul reviewed (changes requested):
see my last comments.

moul commented on docs/docker-compose.yml:
remove this file from this repo.

albttx replied:
This was asked to get a dev env when developing on the docs website to have a preview.

https://github.com/gnolang/gno/pull/1801/files#r1532504955

albttx commented:
Hello @moul , i just updated following your request.

- i moved the docker-compose into `misc/docs.gno.land`
- i replace the `sidebars.js` with a yaml file, see: https://github.com/gnolang/docs.gno.land/pull/22 

wdyt ?

moul commented on misc/docs.gno.land/docker-compose.yml:
```diff
@@ -0,0 +1,27 @@
+version: "3"
```
Why can't you simply move this file to the other repository?

albttx replied:
I'm fine not having it... But i don't know if it's will be really confortable to get easy preview of the docs when working on it...

@leohhhn what do you think ?

moul replied:
Read this: https://github.com/gnolang/gno/pull/1801#discussion_r1581027727

moul commented on docs/sidebars.yml:
```diff
@@ -0,0 +1,146 @@
+---
```
It's not related to the docs. It's barely parseable by Gnoweb.

Please, consider my proposal to use numbered prefixes files and completely remove this file.

If you have valid reasons for not using my suggestion, please create a YAML file for documentation in the "docs/" directory. The file should contain only essential information relevant to docs/, editors, and gnoweb. Let's focus on what truly matters.

Everything related to docs.gno.land should be on the standalone website docs.gno.land, except for the .github/workflows trigger, in case a crontab is insufficient.

albttx replied:
I understand your position, i first update following your comment asking no js, or for json or yaml ( cf: https://github.com/gnolang/gno/pull/1801#discussion_r1580648673 )

But i guess the easiest solutions is to keep a .js file inside the gnolang/docs.gno.land repository and make a PR on the docs.gno.land repository when it's required to update the sidebars.
